lib/bindings/python/rust/lib.rs

```diff
@@ -726,7 +729,15 @@ impl Client {
         let client = self.router.clone();
 
         pyo3_async_runtimes::tokio::future_into_py(py, async move {
-            let stream = client.round_robin(request.into()).await.map_err(to_pyerr)?;
+            let stream = match context {
+                Some(context) => {
+                    let request = RsContext::with_id(request, context.inner().id().to_string());
```

Comment:
Can't you link the new context with the old context before the stream is accepted? I think that would be favorable, e.g. to detect the cancellation during stream setup?

Reply:
Also: making sure: Is the request id, when migrated or sent from a python id related to the orginal id. Would be helpful for tracing.

Reply:
I think this is doable, before the stream is constructed, we can cancel directly at the Controller struct (a.k.a. AsyncEngineContext trait) when it is still owned by the Context struct, with a "thread" monitoring the AsyncEngineContext from the arguments. 

cc @ryanolson if you know there will be issue at the pipeline/network/nats level if we cancel on the Controller struct before a stream is established?